internal/cephfs/fsjournal.go

```diff
@@ -97,6 +120,25 @@ func checkVolExists(ctx context.Context, volOptions *volumeOptions, secret map[s
 	util.DebugLog(ctx, "Found existing volume (%s) with subvolume name (%s) for request (%s)",
 		vid.VolumeID, vid.FsSubvolName, volOptions.RequestName)
 
+	if sID != nil {
+		// This is a work around to fix sizing issue for cloned images
+		err = resizeVolume(ctx, volOptions, cr, volumeID(vid.FsSubvolName), volOptions.Size)
+		if err != nil {
+			klog.Errorf(util.Log(ctx, "failed to expand volume %s: %v"), volumeID(vid.FsSubvolName), err)
+			return nil, err
+		}
+	}
+	if parentVolOpt != nil {
+		err = checkCloneFromSubvolumeExists(ctx, volumeID(pvID.FsSubvolName), volumeID(vid.FsSubvolName), volOptions, parentVolOpt, cr)
+		if err != nil {
+			if errors.As(err, &evnf) {
+				err = j.UndoReservation(ctx, volOptions.MetadataPool,
```

Comment:
`checkCloneFromSubvolumeExists` actually can fail to cleanup intermediate snapshot, due to which clearing the reservation will leak the snapshot as there is no reference left for the same. Such a condition is an internal error and needs to be handled as such without garbage collecting the journal.